Fix performance issue

PhilMFischer opened the pull request:
- Simplified code to Analyse the emf resources and the model
- removed crossreferencer to detect dangling references. was too expensive calculation wise.
- added code which checks all crossreferences from a resource if they are contained instead.
- added a new test case for it
- Adjusted editor to also show emf warnings and not just errors as primary page.

Commit:
Reimplemented resource validation - (Task #509)

- re implemented resource validation
- need to reorganise diagnostics handling
- need to update test cases

---
Task #509: Performance Issue in cross reference check

Commit:
Added squash Diagnostics method to Ecore Helper - (Task #509)

- Added methdod to virsat ecore helper to squash diagnostic messages
- Added test case to it
- Added hamcrest libraries to plugin.

---
Task #509: Performance Issue in cross reference check

Commit:
Interim commit to save current state - (Task #509)

- Cleaned up code for resource analysis in resource set
- still need to update functionality on how to store and deal with
resource and model issues.

---
Task #509: Performance Issue in cross reference check

Commit:
Simplified dangling reference detection - (Task #509)

- Cleaned up code for processing resource analysis.
- Fixed code to correctly detect dangling references.
- Adjusted test case to it
- Changed editor code to also display ProblemsPart as soon as Warnings
show up.


---
Task #509: Performance Issue in cross reference check

Commit:
Fixed test case - (Task #509)

- test case has to expect a warning

---
Task #509: Performance Issue in cross reference check

Commit:
Improved testing - (Task #509)

- Added test case for null content anylysis.

---
Task #509: Performance Issue in cross reference check

Commit:
Added test case - (Task #509)

- Added further test case to check for resource errors and warnings

---
Task #509: Performance Issue in cross reference check

SaMuellerDLR commented on de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/resources/VirSatResourceSet.java:
```diff
@@ -1187,34 +1137,41 @@ public Resource createResource(URI uri) {
+			}
+
+			// Now check severity levels, and set the diagnostics map accordingly
+			// The map will be sued by the editors to display the error messages etc.
```
sued -> used

SaMuellerDLR commented on de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/resources/VirSatResourceSet.java:
```diff
@@ -1224,61 +1181,101 @@ public boolean updateDiagnostic(Resource resource) {
+	}
+	
+	/**
+	 * This method create EMF Diagnostics on the Model Object in case of detected problems
```
create -> creates

SaMuellerDLR commented on de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/resources/VirSatResourceSet.java:
```diff
@@ -1224,61 +1181,101 @@ public boolean updateDiagnostic(Resource resource) {
+								Activator.getPluginId(),
+								0,
+								"Found a dangling reference in due to uncontained object. Press save to fix.",
+								new Object[] { eReferencedObject, eObject }
```
`in due` sentence seems unfinished.

SaMuellerDLR commented on de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/GenericEditor.java:
```diff
@@ -587,7 +587,7 @@ protected void addPages() {
 
 		Diagnostic diagnostic = editingDomain.getResourceSet().getResourceToDiagnosticsMap().get(resource);
-		boolean hasErrors = diagnostic != null && diagnostic.getSeverity() == Diagnostic.ERROR;
+		boolean hasErrors = diagnostic != null && diagnostic.getSeverity() != Diagnostic.OK;
```
Just for clarity: According to the new code I think we would stay on the error page in the case of a warning.
Is this intended?

PhilMFischer replied:
Yes this is intended. We want to react to warnings as well. It is also not staying on the page, but prominently showing it.

SaMuellerDLR replied:
Alright

SaMuellerDLR reviewed (changes requested):
Mostly just minor typos in descriptions.

Commit:
Merge Development

Commit:
Fixed spelling mistakes due to review - (Task #509)

- Fixed grammar in error message
- Fixed some comments

---
Task #509: Performance Issue in cross reference check

SaMuellerDLR reviewed (approved):
Approved!

PhilMFischer commented:
Closes #509
